### PR TITLE
[reporelaypy] Forward branches to alternate remote

### PR DESCRIPTION
**Pull request by JonWBedard**
#### a842447287b58d360960b64d739344648ef6ab4b
<pre>
[reporelaypy] Forward branches to alternate remote
<a href="https://bugs.webkit.org/show_bug.cgi?id=234347">https://bugs.webkit.org/show_bug.cgi?id=234347</a>
&lt;rdar://problem/86526293 &gt;

Reviewed by Dewei Zhu.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/__init__.py: Bump version.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py:
(Checkout.Encoder.default): Add remotes.
(Checkout.clone): After cloning a repository, add required remotes.
(Checkout.add_remotes): Track remotes in repository.
(Checkout.__init__): Allow caller to specify set of remotes to be tracked.
(Checkout.push_update): Push update for specified remote.
(Checkout.update_for): Return &apos;True&apos; and &apos;False.&apos;
(Checkout.update_all): If a branch is being updated, we should forward that
update to any tracked remotes.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/hooks.py:
(HookProcessor.process_worker_hook): After updating a branch, forward that
update to any tracked remotes.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/tests/checkout_unittest.py:
(CheckoutUnittest.test_constructor_no_sentinal): Capture debug logging.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/tests/checkoutroute_unittest.py:
(CheckoutRouteUnittest.test_landing): Capture debug logging.
(CheckoutRouteUnittest.test_invoked_redirect): Ditto.
(CheckoutRouteUnittest.test_trac): Ditto.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/tests/hooks_unittest.py:
(HooksUnittest.test_process): Test for any error logging.
(HooksUnittest.test_process_branch): Ditto.
* Tools/Scripts/libraries/reporelaypy/run: Allow caller to specify remote to be tracked.
* Tools/Scripts/libraries/reporelaypy/setup.py: Bump version.

Canonical link: <a href="https://commits.webkit.org/245300@main">https://commits.webkit.org/245300@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@287115">https://svn.webkit.org/repository/webkit/trunk@287115</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py [hunk 1]
```diff
@@ -60,19 +61,27 @@ def from_json(cls, data):
...
+    @staticmethod
+    def add_remotes(repository, remotes):
+        for name, url in (remotes or {}).items():
+            run([repository.executable(), 'remote', 'add', name, url], cwd=repository.root_path)
+
+    def __init__(self, path, url=None, http_proxy=None, sentinal=True, fallback_url=None, primary=True, remotes=None):
         self.sentinal = sentinal
         self.path = path
         self.url = url
+        self.remotes = dict() or remotes
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py]: Do you mean `self.remotes = remotes or dict()`?

Comment (JonWBedard) [reply Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py]: Yes, this is a pretty silly bug, will fix.

**Review by dewei-zhu — APPROVED** [review]
r=me

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/245300@main (a842447287b58d360960b64d739344648ef6ab4b)!

**Pull request merged**